ldes-server-port-publication-rest/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/rest/converters/LdesFragmentHttpConverter.java

```diff
@@ -53,20 +47,19 @@ public void write(LdesFragment ldesFragment, MediaType contentType, HttpOutputMe
             throws IOException, HttpMessageNotWritableException {
 
         OutputStream body = outputMessage.getBody();
-
-        final RDFFormat rdfFormat;
-        if ("application/n-quads".equals(contentType.toString())) {
-            rdfFormat = NQUADS;
-        } else {
-            rdfFormat = JSONLD11;
-        }
-
+        RDFFormat rdfFormat = getRdfFormat(contentType);
         Model fragmentModel = ldesFragment.toRdfOutputModel();
-
         StringWriter outputStream = new StringWriter();
-
         RDFDataMgr.write(outputStream, fragmentModel, rdfFormat);
-
         body.write(outputStream.toString().getBytes());
     }
+
+    private RDFFormat getRdfFormat(MediaType contentType) {
+        return switch (contentType.toString()) {
+        case "application/n-quads" -> NQUADS;
+        case "application/ld+json" -> JSONLD11;
+        default -> throw new UnsupportedOperationException(
+                String.format("No converter implemented for MediaType %s", contentType));
```

Comment:
I would, by default return JSONLD11, not returning anything might give out the wrong scenario?